src/main/java/com/gepardec/wdg/challenge/validation/URLValidator.java

```diff
@@ -0,0 +1,38 @@
+package com.gepardec.wdg.challenge.validation;
+
+import com.gepardec.wdg.challenge.model.AnswerChallenge2;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+public class URLValidator implements ConstraintValidator<URLValid, AnswerChallenge2> {
+
+    private static final String URL_REGEX = "^((((https?|ftps?|gopher|telnet|nntp)://)|(mailto:|news:))" +
+            "(%[0-9A-Fa-f]{2}|[-()_.!~*';/?:@&=+$,A-Za-z0-9])+)" +
+            "([).!';/?:,][[:blank:]])?$";
```

Comment:
Die regex kann simpler dargestellt werden

Reply:
Wir sind davon ausgegangen, dass man zuerst prüft, ob die übergebene Antwort überhaupt ein URL ist (mittels `URLValidator`) und danach änlich wie beim Challenge 1, prüfen wir die Richtigkeit der Antwort gegen `CHALLENGE2.getAnswer()`. Ich mache es grad anders aber wollte nur hier nochmal hinschreiben, falls es anders interpretiert wurde (weil genau das, was hier beschrieben wurde, wird in der `Zeile 95 - ChallengesResource.java` gemacht)